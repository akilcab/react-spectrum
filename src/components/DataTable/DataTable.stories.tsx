import { Story, Meta } from "@storybook/react/types-6-0";
import { DataTable } from "./DataTable";
import { Paper, TableContainer } from "@material-ui/core";

export default {
  title: "DataTable",
  component: DataTable,
  description: `A DataTable.`,
  argTypes: {
    rows: Array,
    columns: Array,
  },
} as Meta;

//👇 We create a “template” of how args map to rendering
const Template: Story = (args) => (
  <TableContainer component={Paper} elevation={0} style={{ height: "38rem" }}>
    <DataTable columns={[]} rows={[]} {...args} />
  </TableContainer>
);

//👇 Each story then reuses that template
export const Default = Template.bind({});
Default.args = {
  columns: [],
  rows: [],
};
