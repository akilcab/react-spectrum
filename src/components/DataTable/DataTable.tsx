import { DataGrid } from "@material-ui/data-grid";
import { createStyles, withStyles } from "@material-ui/core";

export const DataTable = withStyles(() =>
  createStyles({
    root: {
      border: "none",
      "& .MuiDataGrid-row.Mui-even:not(:hover)": {
        backgroundColor: "#ebebeb",
      },
      "& .MuiDataGrid-row": {
        backgroundColor: "#fff",
      },
      "& .MuiDataGrid-footerContainer": {
        backgroundColor: "none",
      },
      "& .MuiDataGrid-columnsContainer": {
        backgroundColor: "#fff",
        borderRadius: "8px",
        border: "1px solid #fafafa",
      },
      "& .MuiDataGrid-columnSeparator": {
        display: "none",
      },
      "& .MuiDataGrid-cell": {
        borderBottom: "none",
      },
    },
  })
)(DataGrid);
