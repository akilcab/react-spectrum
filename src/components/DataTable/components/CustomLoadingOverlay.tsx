import { LinearProgress } from "@material-ui/core";
import { GridOverlay } from "@material-ui/data-grid";

export default () => (
  <GridOverlay>
    <div style={{ position: "absolute", top: 0, width: "100%" }}>
      <LinearProgress />
    </div>
  </GridOverlay>
);
