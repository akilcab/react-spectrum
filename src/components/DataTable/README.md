# Description of component

Data tables display information in a grid-like format of rows and columns. They organize information in a way that’s
easy to scan so that users can look for patterns and insights.

The component leverages the power of React and TypeScript, to provide the best UX while manipulating an unlimited set of
data. It comes with an intuitive API for real-time updates, accessibility, as well as theming and custom templates, all
with blazing fast performance.

#### Example :

![](assets/desc.png)

## How to import

```node
import { DataTable } from "app/shared/components/DataTable";
```

## How to use

```node
<TableContainer component={Paper} elevation={0} style={{ height: "38rem" }}>
  <DataTable
    columns={columnsAccountingPlan}
    rows={data}
    loading={isLoading}
    checkboxSelection
    disableSelectionOnClick
    density="compact"
    disableColumnMenu
    rowsPerPageOptions={[20, 50, 100]}
    pageSize={20}
    components={{
      LoadingOverlay: CustomLoadingOverlay,
    }}
    localeText={{
      errorOverlayDefaultLabel: "Une erreur s'est produite.",
      noRowsLabel: "Aucun résultat trouvé.",
      toolbarDensity: "Compact",
    }}
    hideFooterSelectedRowCount={true}
  />
</TableContainer>
```

## Props

All others DataGrid Community props

## CSS properties we can override

You can override all CSS properties of the DataGrid Community type of Material UI.
