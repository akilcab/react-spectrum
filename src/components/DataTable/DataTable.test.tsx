import renderer from "react-test-renderer";
import { Paper, TableContainer } from "@material-ui/core";
import CustomLoadingOverlay from "./components/CustomLoadingOverlay";
import { DataTable } from "./DataTable";

describe("<DataTable />", () => {
  test("rend correctement", () => {
    const component = renderer
      .create(
        <TableContainer
          component={Paper}
          elevation={0}
          style={{ height: "38rem" }}
        >
          <DataTable
            columns={[
              {
                field: "field",
                headerName: "col",
                flex: 1,
              },
            ]}
            rows={[
              {
                id: "row1",
                field: "row",
              },
            ]}
            loading={false}
            checkboxSelection
            disableSelectionOnClick
            density="compact"
            disableColumnMenu
            rowsPerPageOptions={[20, 50, 100]}
            pageSize={20}
            components={{
              LoadingOverlay: CustomLoadingOverlay,
            }}
            localeText={{
              errorOverlayDefaultLabel: "Une erreur s'est produite.",
              noRowsLabel: "Aucun résultat trouvé.",
              toolbarDensity: "Compact",
            }}
            hideFooterSelectedRowCount={true}
          />
        </TableContainer>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
