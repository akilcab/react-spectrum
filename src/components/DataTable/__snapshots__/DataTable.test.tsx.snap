// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTable /> rend correctement 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation0 MuiPaper-rounded"
  style={
    Object {
      "height": "38rem",
    }
  }
/>
`;
